### PR TITLE
Add gsw indicators

### PR DESCRIPTION
**Pull request by karpfen**
This PR adds a couple of Global Surface Water indicators

### CODE REVIEW

File: R/calc_gsw_change.R [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#' The pixel values are aggregated using method provided via the
+#' \code{stats_gsw} parameter.
+#'
+#' @param x A single polygon for which to calculate the GSW statistics.
+#' @param global_surface_water_change The GSW Change data source.
+#' @param engine The preferred processing functions from either one of "zonal",
+#' "extract" or "exactextract". Default: "extract".
+#' @param stats_gsw Aggregation function with which the data are combined.
+#' Default: "mean".
+#' @return A tibble
```
Comment (goergen95) [inline R/calc_gsw_change.R:20]: Could you please describe briefly columns names and what they mean here? (applies to all indicators)

File: R/calc_gsw_occurrence.R [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#' The pixel values are aggregated using method provided via the
+#' \code{stats_gsw} parameter.
+#'
+#' @param x A single polygon for which to calculate the GSW statistics.
+#' @param global_surface_water_occurrence The GSW Occurrence data source.
+#' @param engine The preferred processing functions from either one of "zonal",
+#' "extract" or "exactextract". Default: "extract".
+#' @param stats_gsw Aggregation function with which the data are combined.
+#' Default: "mean".
+#' @return A tibble
```
Comment (goergen95) [inline R/calc_gsw_occurrence.R:18]: see above

File: R/calc_gsw_recurrence.R [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#' The pixel values are aggregated using method provided via the
+#' \code{stats_gsw} parameter.
+#'
+#' @param x A single polygon for which to calculate the GSW statistics.
+#' @param global_surface_water_recurrence The GSW Recurrence data source.
+#' @param engine The preferred processing functions from either one of "zonal",
+#' "extract" or "exactextract". Default: "extract".
+#' @param stats_gsw Aggregation function with which the data are combined.
+#' Default: "mean".
+#' @return A tibble
```
Comment (goergen95) [inline R/calc_gsw_recurrence.R:19]: see above

File: R/calc_gsw_seasonality.R [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+#' The pixel values are aggregated using method provided via the
+#' \code{stats_gsw} parameter.
+#'
+#' @param x A single polygon for which to calculate the GSW statistics.
+#' @param global_surface_water_seasonality The GSW Seasonality data source.
+#' @param engine The preferred processing functions from either one of "zonal",
+#' "extract" or "exactextract". Default: "extract".
+#' @param stats_gsw Aggregation function with which the data are combined.
+#' Default: "mean".
+#' @return A tibble
```
Comment (goergen95) [inline R/calc_gsw_seasonality.R:16]: see above

File: R/calc_gsw_transitios.R
Comment (goergen95) [inline R/calc_gsw_transitios.R:1]: typo in filename

File: R/calc_gsw_transitios.R [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+#' | 7     | Seasonal to Permanent |
+#' | 8     | Permanent to Seasonal |
+#' | 9     | Ephemeral Permanent   |
+#' | 10    | Ephemeral Seasonal    |
+#'
+#' To aggregate, we sum up the area of each transition class for a given region.
+#'
+#' @param x A single polygon for which to calculate the GSW statistics.
+#' @param global_surface_water_transitions The GSW Transitions data source.
+#' @return A tibble
```
Comment (goergen95) [inline R/calc_gsw_transitios.R:24]: see above

File: R/calc_gsw_transitios.R [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  )
+)
+
+register_indicator(
+  name = "gsw_transitions",
+  resources = list(global_surface_water_transitions = "raster"),
+  fun = .calc_gsw_transitions,
+  arguments = list(
+    engine = "extract",
+    stats_gsw = "mean"
```
Comment (goergen95) [inline R/calc_gsw_transitios.R:69]: argument `stats_gsw` needs to be removed here

File: R/get_gsw.R
Comment (goergen95) [inline R/get_gsw.R:1]: please include check if the supplied statistic and version is valid and throw an error if any of the is not

File: R/get_gsw.R [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  }
+  if (max_y < 0) {
+    max_y <- paste0(sprintf(formatstr_y, abs(max_y)), "S")
+  } else {
+    max_y <- paste0(sprintf(formatstr_y, abs(max_y)), "N")
+  }
+  paste0(min_x, "_", max_y)
+}
+
+gsw_statistics <- c(
```
Comment (goergen95) [inline R/get_gsw.R:62]: please add a leading point `.gsw_statistics` to indicate that the object is internal

File: inst/res/gsw_change/change_60W_10Nv1_4_2021_.tif.aux.xml
Comment (goergen95) [inline inst/res/gsw_change/change_60W_10Nv1_4_2021_.tif.aux.xml:1]: i would prefer not to include the xml files here. Also, please provide a processing script in `data-raw` and make sure that the tifs are as small as possible (e.g. set `COMPRESS=LZW` and potentially decrease resolution if to large)

File: tests/testthat/test-calc_gsw_seasonality.R [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    system.file("extdata", "shell_beach_protected_area_41057_B.gpkg",
+      package = "mapme.biodiversity"
+    )
+  )
+
+  shp <- suppressWarnings(st_cast(shp, to = "POLYGON")[1, ])
+  portfolio <- init_portfolio(
+    shp,
+    years = 2024,
+    tmpdir =  "~/R/mapme.biodiversity/inst/res/gsw_seasonality",
```
Comment (goergen95) [inline tests/testthat/test-calc_gsw_seasonality.R:12]: see above

**Review by goergen95 — CHANGES_REQUESTED** [review]
Great work and really interesting indicators. Would love to have a chat with you on how these are used in practice. Please see my comments, some minor changes are required before mergeing.

**Comment by karpfen** [comment]
Thanks for your input @goergen95, that was really helpful! All points above are addressed now.

File: inst/res/gfw_treecover/Hansen_GFC-2020-v1.8_treecover2000_20N_080W.tif.aux.xml
Comment (goergen95) [inline inst/res/gfw_treecover/Hansen_GFC-2020-v1.8_treecover2000_20N_080W.tif.aux.xml:1]: I thinks this slipped through somehow. Could you please remove it?

**Review by goergen95 — APPROVED** [review]

**Pull request merged**